### DISCUSSION TITLE
Host key verification failed.

### DISCUSSION THREAD

**Discussion by ngohuytrieu** [community/community#27716] [Actions] [answered]
I am trying to use Github Actions SSH to AWS EC2 but I get this error:
```
Run ssh -i ~/.ssh/private.key -tt username@*** "
Host key verification failed.
Error: Process completed with exit code 255.
```
Here is my Github actions
```
- name: Create SSH key
        run: |
          mkdir -p ~/.ssh/
          echo "$SSH_PRIVATE_KEY" > ~/.ssh/private.key
          sudo chmod 600 ~/.ssh/private.key
          echo "$SSH_KNOWN_HOSTS" > ~/.ssh/known_hosts
          sudo chmod 600 ~/.ssh/known_hosts
        shell: bash
        env:
          SSH_PRIVATE_KEY: ${{secrets.SSH_KEY}}
          SSH_KNOWN_HOSTS: ${{secrets.SSH_HOSTNAME}}
      - name: SSH into EC2
        run: >
          ssh -i ~/.ssh/private.key -tt username@hostname
          "
            echo 'Welcome to EC2'
          "
```

@airtower-luna I've just created a new discussion for this topic

**Comment by airtower-luna** [comment, marked as answer]
In principle there are three things that can cause this:

* Something or someone is interfering with your connection. Unlikely, would be bad if true, but catching it is why we need to check host keys.
* The `known_hosts` file isn't correct, e.g. the public key for the host you want isn't the right one, or otherwise broken. I can't see what you're writing to it so I can't really say, but that the secret that gets written to it is called `SSH_HOSTNAME` makes me a bit suspicious, because writing just the hostname won't work, you need to include the expected public key of the server.
* Assuming the public key for the server in the `known_hosts` is correct, there could be another problem: Maybe the public key type negotiated by the SSH server and client isn't the one of the key you put into the `known_hosts` file. SSH servers usually have *multiple* public keys of different types (e.g. RSA, ed25519). Possibly the server is trying to use another (correct) one.

To figure out what's going on:

* Run `ssh` with `-v` to get debug output. Look which algorithms are negotiated, which keys are used, and if there are any issues with the `known_hosts` file.
* Check the actual content of the `known_hosts` file. It's easiest if have a good one locally, in that case you could just copy the relevant line.

**Reply by ngohuytrieu** [reply]
Thank @airtower-luna turned out my known_hosts is in an incorrect format.  I copied the EC2 public DNS instead of known_hosts.
I changed it into known_hosts then it worked now. Thank you.

**Reply by guettli** [reply]
Check that `$HOME` is correct by comparing `getent passwd $(id -un) | cut -d: -f6` with `$HOME`. You might be affected by: [HOME is overridden in Container](https://github.com/actions/runner/issues/863)
